src/util/android/scheduler.hpp

```diff
@@ -115,6 +115,11 @@ class ALooperScheduler : public util::Scheduler, public std::enable_shared_from_
     {
         return m_thread == pthread_self();
     }
+    bool is_same_as(const Scheduler* other) const noexcept override
+    {
+        const ALooperScheduler* o = dynamic_cast<const ALooperScheduler*>(other);
```

Comment:
This is a good place to use `auto` as the type name already appears in this line on the rhs.